src/main/scala/org/broadinstitute/dsde/vault/services/generic/GenericIngestService.scala

```diff
@@ -0,0 +1,57 @@
+package org.broadinstitute.dsde.vault.services.generic
+
+import com.wordnik.swagger.annotations._
+import org.broadinstitute.dsde.vault.common.directives.VersioningDirectives._
+import org.broadinstitute.dsde.vault.model.{GenericSysAttrs, GenericEntity, GenericIngest}
+import org.broadinstitute.dsde.vault.model.GenericJsonProtocol._
+import spray.json._
+import spray.http.MediaTypes._
+import spray.httpx.SprayJsonSupport._
+import spray.routing.HttpService
+
+@Api(value="/entities", description="generic entity service", produces="application/json")
+trait GenericIngestService extends HttpService {
+  private final val ApiPrefix = "entities"
+  private final val ApiVersions = "v1"
+  private final val DefaultVersion = 1
+
+  val giRoute = ingestRoute
+
+  @ApiOperation(
+    value = "store some entities and relationships",
+    nickname = "genericIngest",
+    httpMethod = "POST",
+    response = classOf[GenericEntity],
+    responseContainer = "List",
+    notes = "response is a list of the new entities")
```

Comment:
DM is just returning the UUIDs it created, not the whole entities. From the stub below, it looks like you'll handle returning the whole entities (and relations) by replacing the relevant parts of the request. If this works, great! If that's difficult, it might require GET-ing the entities again ... extra queries. No action necessary from this comment ... just tossing it out there.